src/bigint.rs

```diff
@@ -467,146 +522,303 @@ impl Unsigned for BigUint {}
 
 forward_all_binop_to_val_ref_commutative!(impl Add for BigUint, add);
 
-impl<'a> Add<&'a BigUint> for BigUint {
+fn add2(a: &mut [BigDigit], b: &[BigDigit]) {
+    let mut b_iter = b.iter();
+    let mut carry = 0;
+
+    for ai in a.iter_mut() {
+        if let Some(bi) = b_iter.next() {
+            *ai = adc(*ai, *bi, &mut carry);
+        } else if carry != 0 {
+            *ai = adc(*ai, 0, &mut carry);
+        } else {
+            break;
+        }
+    }
+
+    assert!(b_iter.next() == None && carry == 0);
+}
+
+impl<'a> Add<&'a [BigDigit]> for BigUint {
```

Comment:
I don't know any way to hide it.  This is a public trait implemented for public types -- the result is public.  I suspect if `BigDigit` weren't public, it might stay hidden... but I'm not sure since it's a type alias and not a newtype.

I've tinkered in small ways on the compiler, but nothing like what you're asking for. :)